File: 2015/ruby/day1.rb
data = File.read('day1.txt')
floor = 0
data.chars.each_with_index do |c, i| 
	if c == '('; 
		floor +=1; 
	elsif c == ')'; 
		floor -= 1; 
	end
  if floor == -1; puts i+1; end
end
puts floor

File: 2015/go/day1/day1.go
package day1

func floor(input string) int {
	var floor = 0
	for i := 0; i < len(input); i++ {
		var c = input[i]
		if c == '(' {
			floor = floor + 1
		} else if c == ')' {
			floor = floor - 1
		}
	}
	return floor
}

func basement(input string) int {
	var floor = 0
	for i := 0; i < len(input); i++ {
		var c = input[i]
		if c == '(' {
			floor = floor + 1
		} else if c == ')' {
			floor = floor - 1
		}
		if floor == -1 {
			return i + 1
		}
	}
	return floor
}
